libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/impl/CertificateIdentityParser.java

```diff
@@ -0,0 +1,166 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+import com.yahoo.athenz.auth.util.AthenzUtils;
+import com.yahoo.athenz.auth.util.Crypto;
+
+/**
+ * Parser of Athenz identity
+ */
+public class CertificateIdentityParser {
+
+    /**
+     * X509Certificate attribute name
+     */
+    public static final String JAVAX_CERT_ATTR = "javax.servlet.request.X509Certificate";
+
+    /**
+     * Role prefix inside X509Certificate
+     */
+    public static final String ZTS_CERT_ROLE_URI = "athenz://role/";
+
+    public static final String EMPTY_CERT_ERR_MSG = "No certificate available in request";
+
+    private Set<String> excludedPrincipalSet = null;
+    private boolean excludeRoleCertificates;
+
+    /**
+     * @param  excludedPrincipalSet    Reject parsing certificate with those principal
+     * @param  excludeRoleCertificates Reject parsing role certificates
+     */
+    public CertificateIdentityParser(Set<String> excludedPrincipalSet, boolean excludeRoleCertificates) {
+        this.excludedPrincipalSet = excludedPrincipalSet;
+        this.excludeRoleCertificates = excludeRoleCertificates;
+    }
+
+    /**
+     * Parse from X509Certificate inside the request.
+     * @param  request                      HTTPS request
+     * @return                              CertificateIdentity
+     * @throws CertificateIdentityException parse error
+     */
+    public CertificateIdentity parse(HttpServletRequest request) throws CertificateIdentityException {
+        X509Certificate[] certs = (X509Certificate[]) request.getAttribute(JAVAX_CERT_ATTR);
+        return this.parse(certs);
+    }
+
+    /**
+     * Parse from X509Certificate chain.
+     * @param  certs                        X509Certificate chain
+     * @return                              CertificateIdentity
+     * @throws CertificateIdentityException parse error
+     */
+    public CertificateIdentity parse(X509Certificate[] certs) throws CertificateIdentityException {
+        // make sure we have at least one valid certificate in our list
+
+        if (certs == null || certs[0] == null) {
+            throw new CertificateIdentityException(EMPTY_CERT_ERR_MSG);
+        }
+
+        X509Certificate x509Cert = certs[0];
+        String principalName = Crypto.extractX509CertCommonName(x509Cert);
+        if (principalName == null || principalName.isEmpty()) {
+            throw new CertificateIdentityException("Certificate principal is empty");
+        }
+
+        // make sure the principal is not on our excluded list
+
+        if (this.excludedPrincipalSet != null && this.excludedPrincipalSet.contains(principalName)) {
+            throw new CertificateIdentityException("Principal is excluded");
+        }
+
+        // For role cert, the principal information is in the SAN email
+
+        List<String> roles = null;
+        int idx = principalName.indexOf(":role.");
```

Comment:
we have :role. defined in AuthortyConsts so we should use that here

Reply:
Updated in 2c5c855c, and try to refactor other hard-coded `:role.` too.

hard-coded `:role.` values are preserved in these files, as their context seem to be different.
```
./servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSResources.java
./servers/zts/src/main/java/com/yahoo/athenz/zts/AccessTokenRequest.java
./servers/zts/src/main/java/com/yahoo/athenz/zts/store/DataStore.java
*Test.java
```